osf/migrations/0084_preprint_node_divorce.py

```diff
@@ -0,0 +1,69 @@
+# -*- coding: utf-8 -*-
+# Generated by Django 1.11.9 on 2018-03-12 18:25
+from __future__ import unicode_literals
+
+from django.db import migrations
+from django.db.models import F
+from itertools import islice
+
+
+def divorce_preprints_from_nodes(apps, schema_editor):
+    Preprint = apps.get_model('osf', 'PreprintService')
```

Comment:
While I know that importing models this way in migrations is preferred, the Preprint model doesn't seem to have access to the methods in the Guardian mixin, probably related to https://github.com/CenterForOpenScience/osf.io/pull/8117#discussion_r165541392.   So calling preprint.get_group below won't work.   I would try directly importing the PreprintService model for now just to get this to run all the way through.

Another solution, maybe import the Group model directly and not rely on the functions in the Guardian mixin.   So instead of preprint.get_group('admin'), do  Group.objects.get(name=self.format_group(name)).

Reply:
Prefer `Group` model import -- won't require as much refactoring later